pkg/reconciler/route/domains/domains.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package domains
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/knative/serving/pkg/network"
+	"github.com/knative/serving/pkg/reconciler/route/config"
+)
+
+// HTTPScheme is the string representation of http.
```

Comment:
Orthogonal question: is this going to be `http` forever, or we're going to permit `https`? If the latter, we might want to use a enum somewhere in `pkg/network`. or `pkg/apis/networking/register.go`.

Reply:
Not quite sure, however at the time of implementation this was the only supported mode it seems.

https://github.com/knative/serving/blob/33da2843953b6b01414504ea2d10a3c751e612f3/pkg/reconciler/route/traffic/traffic.go#L35-L36

Reply:
These are going to collide: https://github.com/knative/serving/pull/3902